sale_delivery_date/models/sale_order_line.py

```diff
@@ -6,124 +6,201 @@
 
 import pytz
 
-from odoo import api, fields, models
+from odoo import _, api, exceptions, fields, models
 
 from odoo.addons.partner_tz.tools import tz_utils
 
 _logger = logging.getLogger(__name__)
 
+FIND_WORKING_DAY_COUNT = 10  # To avoid infinite loops, could be increased
+
 
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
+    def _action_launch_stock_rule(self, previous_product_uom_qty=False):
+        # Add a context key, so we know in `_prepare_procurement_values` that we
+        # might have to update the date based on today instead of the order date
+        if previous_product_uom_qty:
+            self = self.with_context(updated_line_qty=True)
+        return super(SaleOrderLine, self)._action_launch_stock_rule(
+            previous_product_uom_qty
+        )
+
     def _prepare_procurement_values(self, group_id=False):
+        # Update 'date_planned' and 'date_deadline' with respect to:
+        #   - the warehouse or partner cutoff time
+        #   - the warehouse calendar
+        #   - the delivery time window of the customer
+        # knowing that 'date_planned' is when the transfer starts to be processed
+        # internally and 'date_deadline' is when the goods will be delivered to
+        # the customer.
+        # There's 2 cases here:
+        #   1) commitment_date is set, compute date_planned from date_deadline
+        #   2) commitment_date isn't set, compute date_planned and date_deadline
         res = super()._prepare_procurement_values(group_id=group_id)
-        res = self._cutoff_time_delivery_prepare_procurement_values(res)
-        res = self._warehouse_calendar_prepare_procurement_values(res)
-        res = self._delivery_window_prepare_procurement_values(res)
+        if self.order_id.commitment_date:
```

Comment:
As you now depend on `sale_order_line_date`
```suggestion
        if self.commitment_date or self.order_id.commitment_date:
```